From 45c1d4fc6ed09d6e1a5412c2ac902b236b4e212f Mon Sep 17 00:00:00 2001
From: Eygene Ryabinkin <rea@codelabs.ru>
Date: Sun, 19 Feb 2012 20:55:06 +0400
Subject: [PATCH] Fix Curses interface for Python 2.6

In 2.6 all logging classes are not the new-style ones, so they
have no mro() method and, thus, we can't use super() for them.
Since CursesLogHanler is singly-inherited method, there will
be no problems in usage of the explicit superclass name.

Signed-off-by: Eygene Ryabinkin <rea@codelabs.ru>
---
 offlineimap/ui/Curses.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/offlineimap/ui/Curses.py b/offlineimap/ui/Curses.py
index af41d35..fa791bb 100644
--- a/offlineimap/ui/Curses.py
+++ b/offlineimap/ui/Curses.py
@@ -282,7 +282,7 @@ class CursesLogHandler(logging.StreamHandler):
     """self.ui has been set to the UI class before anything is invoked"""
 
     def emit(self, record):
-        log_str = super(CursesLogHandler, self).format(record)
+        log_str = logging.StreamHandler.format(self, record)
         color = self.ui.gettf().curses_color
         # We must acquire both locks.  Otherwise, deadlock can result.
         # This can happen if one thread calls _msg (locking curses, then
-- 
1.7.9

